romdisk/date.asm

```diff
@@ -13,25 +13,63 @@
         EXTERN init_static_buffer
         DEFC STATIC_DATE_BUFFER = init_static_buffer
 
-        ; date main routine. 
-        ; Print the current date and time on screen
+        ; Date main routine, print the current date and time on screen.
+        ; With zero arguments, just print date.
+        ; With one argument, set date; with two arguments, set date and time.
+        ; Date and time format is:
+        ;       YYYY-MM-DD HH:MM:SS
         ; Parameters:
         ;       HL - ARGV
         ;       BC - ARGC
         ; Returns:
         ;       A - 0 on success
         PUBLIC date_main
 date_main:
-        ; Ignore parameters
+        ; Read the existing date whether we are reading or writing
+        ld de, STATIC_DATE_BUFFER
         ; Get system date:
         ;   DE - Address of the date structure to fill. Must be at least DATE_STRUCT_SIZE bytes.
         ; Returns:
         ;   A - ERR_SUCCESS on success, error code else
-        ld de, STATIC_DATE_BUFFER
+        push hl
         GETDATE()
```

Comment:
I think this is not necessary if C is > 1 since we would try to set the date in that case.
Here we should perform:
```
    dec c
    jr z, _date_main_print
    [...]
_date_main_print:
    ld de, STATIC_DATE_BUFFER
    GETDATE()
```

Reply:
The reason for doing the clock read even when setting the clock is that we have the option to just set the date without setting the time - so the read fills out the time to that it doesn't get reset.

Reply:
@chiralos Oh you are right, I forgot about that!